api/internal/localizer/localizer.go

```diff
@@ -0,0 +1,364 @@
+// Copyright 2022 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package localizer
+
+import (
+	"log"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"sigs.k8s.io/kustomize/api/ifc"
+	"sigs.k8s.io/kustomize/api/internal/git"
+	pluginsLoader "sigs.k8s.io/kustomize/api/internal/plugins/loader"
+	"sigs.k8s.io/kustomize/api/internal/target"
+	"sigs.k8s.io/kustomize/api/konfig"
+	"sigs.k8s.io/kustomize/api/krusty"
+	"sigs.k8s.io/kustomize/api/loader"
+	"sigs.k8s.io/kustomize/api/provider"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+	"sigs.k8s.io/yaml"
+)
+
+// localizer localizes the kustomization root at ldr.
+type localizer struct {
+	fSys filesys.FileSystem
+
+	ldr ifc.Loader
+
+	// destination directory in newDir that mirrors ldr's current root.
+	dst filesys.ConfirmedDir
+
+	// all localize directories created
+	localizeDirs map[filesys.ConfirmedDir]struct{}
+
+	// kusttarget fields
+	rFactory  *resmap.Factory
+	pLdr      *pluginsLoader.Loader
+	validator ifc.Validator
+}
+
+// Run attempts to localize the kustomization root at targetArg with the given localize arguments
+func Run(fSys filesys.FileSystem, targetArg string, scopeArg string, newDirArg string) error {
+	ldr, args, err := NewLocLoader(targetArg, scopeArg, newDirArg, fSys)
+	if err != nil {
+		return err
+	}
+	defer ldr.Cleanup()
+
+	// create directory that corresponds to target in newDir
+	var realScope string
+	if repo, isRemote := ldr.Repo(); isRemote {
+		realScope = repo
+	} else {
+		realScope = args.Scope.String()
```

Comment:
WDYT about moving this if/else logic into `NewLocLoader`? That is, in the case of remote, set `args.Scope` to the `ldr.Repo()`?

Reply:
Good idea! Done.